jans-cedarling/cedarling/src/jwt/decoding_strategy/key_service.rs

```diff
@@ -9,32 +9,118 @@ mod error;
 mod openid_config;
 
 pub use error::KeyServiceError;
-use jsonwebtoken::jwk::JwkSet;
+use jsonwebtoken::jwk::Jwk;
 use jsonwebtoken::DecodingKey;
 pub(crate) use openid_config::*;
 use reqwest::blocking::Client;
+use serde::Deserialize;
 use std::collections::HashMap;
 use std::sync::Arc;
+use std::thread::sleep;
+use std::time::Duration;
 
+/// A wrapper around `reqwest::blocking::Client` providing HTTP request functionality with retry logic.
+///
+/// The `HttpClient` struct allows for sending GET requests with a retry mechanism that attempts to
+/// fetch the requested resource up to a maximum number of times if an error occurs.
+pub struct HttpClient {
+    client: reqwest::blocking::Client,
+    max_retries: u32,
+    retry_delay: Duration,
+}
+
+impl HttpClient {
+    pub fn default() -> Result<Self, KeyServiceError> {
```

Comment:
how about implement `Default` trait? I don't know if it good idea but consider

Reply:
i tried that but i can't return a `Result` from the `Default` trait implementation